hibernate-core/src/main/java/org/hibernate/cfg/OneToOneSecondPass.java

```diff
@@ -237,18 +237,6 @@ else if ( otherSideProperty.getValue() instanceof ManyToOne ) {
 				boolean referenceToPrimaryKey  = referencesDerivedId || mappedBy == null;
 				value.setReferenceToPrimaryKey( referenceToPrimaryKey );
 
-				// If the other side is an entity with an ID that is derived from
-				// this side's owner entity, and both sides of the association are eager,
-				// then this side must be set to FetchMode.SELECT; otherwise,
-				// there will be an infinite loop attempting to load the derived ID on
-				// the opposite side.
-				if ( referencesDerivedId &&
```

Comment:
I guess this is now handled by `adjustJoinFetchIfNeeded`?

Reply:
@beikov, yes, that's correct.